src/CommitmentsV2/SFA.DAS.CommitmentsV2.UnitTests/TestHelper.cs

```diff
@@ -14,5 +18,11 @@ public static T Clone<T>(T source)
             var serialized = JsonConvert.SerializeObject(source, settings);
             return JsonConvert.DeserializeObject<T>(serialized, settings);
         }
+
+        public static ProviderCommitmentsDbContext GetInMemoryDatabase() => new ProviderCommitmentsDbContext(new DbContextOptionsBuilder<ProviderCommitmentsDbContext>()
+            .UseInMemoryDatabase(Guid.NewGuid().ToString())
```

Comment:
I've found we get intermittent test failures sometimes when tests run in parallel, and I _think_ using the following overload of `.UseInMemoryDatabase` fixes it:

`.UseInMemoryDatabase(Guid.NewGuid().ToString(), new InMemoryDatabaseRoot())`

I have paired that elsewhere with the following tear down method in the fixture:

```
           public void TearDown()
            {
                Db.Database.EnsureDeleted();
            }
```
invoked like so:

```
        [TearDown]
        public void TearDown()
        {
            _fixture.TearDown();
            _fixture = null;
        }
```
See `CohortDomainServiceTests.cs`, which used to fail 50% of the time.

Reply:
By the way I'm not implying that there are any errors running your new set of tests in parallel. But this might be worth adding so that it stays that way.

Reply:
Tried adding the TearDown but it throws a disposed exception when it reaches
```
                Db.Database.EnsureDeleted();
```
as UlnUtilisationService.GetUlnUtilisations() wraps context in using statement and disposes prior to teardown, also tried:
```
                Db?.Database?.EnsureDeleted();
```
but got same error, added the other changes though

Reply:
Ok, I think that will be fine. It was belt and braces anyway.